File: client/src/ts/frame.ts
import { DrawingTool, Point } from "./types";

interface FrameData {
  drawingTool: string;
  points?: Point[];
  createdAt?: Date;
}

export class Frame {
  public drawingTool: string;
  public points: Point[];
  public createdAt: Date;

  constructor({ drawingTool, points = [], createdAt = new Date() }: FrameData) {
    this.drawingTool = drawingTool;
    this.points = points;
    this.createdAt = createdAt;
  }

  public add(x: number, y: number) {
    this.points.push([x, y]);
    return this;
  }
}

File: server/src/paper/frame.rs
extern crate chrono;

// use chrono::{DateTime, Utc};

use crate::paper::point::Point;

#[derive(Serialize, Deserialize, Debug)]
pub(crate) struct Palette {
    pub name: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub(crate) struct Frame {
    // pub created_at: DateTime<Utc>,
    pub points: Vec<Point>,
    pub drawing_tool: String,
    pub palettes: Vec<Palette>,
}

impl Frame {
    pub fn new() -> Frame {
        Frame {
            drawing_tool: String::from("brush"),
            palettes: vec![],
            // created_at: Utc::now(),
            points: vec![],
        }
    }
}
